webknossos/webknossos/dataset/layer.py

```diff
@@ -372,50 +371,41 @@ def delete_mag(self, mag: Union[int, str, list, tuple, np.ndarray, Mag]) -> None
         rmtree(full_path)
 
     def _add_foreign_mag(
-        self, foreign_mag_path: Path, symlink: bool, make_relative: bool
+        self, foreign_mag_view: MagView, symlink: bool, make_relative: bool
     ) -> MagView:
-        mag_name = foreign_mag_path.name
-        mag = Mag(mag_name)
-        operation = "symlink" if symlink else "copy"
-        if mag in self.mags.keys():
-            raise IndexError(
-                f"Cannot {operation} {foreign_mag_path}. This dataset already has a mag called {mag_name}."
-            )
+        is_foreign_mag_compressed = (
+            foreign_mag_view.header.block_type != wkw.Header.BLOCK_TYPE_RAW
+        )
+        added_mag_view = self.add_mag(
+            foreign_mag_view.mag,
+            foreign_mag_view.header.block_len,
+            foreign_mag_view.header.file_len,
+            is_foreign_mag_compressed,
+        )
```

Comment:
This might be problematic, since there are multiple compression possibilities and the mag object caches the header.

Reply:
PS: I've added `add_mag_for_existing_files` in the REST API work, this might help here. I'll open a separate PR for it, so you might already use it.

Reply:
Perfect! Thanks!